app/src/main/java/ir/mirrajabi/pc/sample/MainActivity.java

```diff
@@ -59,4 +64,22 @@ public void onClick(PersianDate date) {
         calendar.setColorBackground(getResources().getColor(android.R.color.holo_blue_dark));
         persianCalendarView.update();
     }
+
+    @Override
+    public void onAttachFragment(Fragment fragment) {
+        super.onAttachFragment(fragment);
+        setFarsiLang();
+    }
+
+    private void setFarsiLang(){
+        Locale locale = new Locale("FA");
+        Locale.setDefault(locale);
+        Configuration configuration = getBaseContext().getResources().getConfiguration();
+        configuration.locale = locale;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+            configuration.setLayoutDirection(locale);
+        }
+        getBaseContext().getResources().updateConfiguration(configuration, getBaseContext().getResources().getDisplayMetrics());
+        recreate();
+    }
```

Comment:
This is considered as a bad practice in general. if you want to do it in your app, feel free. but It's not going to be there in the sample app.